attributes/network/cddl/9_account.cddl

```diff
@@ -0,0 +1,182 @@
+    0 => [ * account-role ],
+}
+; end::listRoles[]
+
+; tag::getRoles[]
+; List the roles of an identity.
+account.getRoles@param = {
+    ; The account to get the roles of.
+    0 => account-identity,
+
+    ; The identity to get the roles from.
+    1 => non-anonymous-identity / [ * non-anonymous-identity ],
+}
+
+account.getRoles@return = {
+    ; The list of roles, per identity.
+    0 => {
+        identity => [ * account-role ],
+    }
+}
+; end::getRoles[]
+
+; tag::addRoles[]
+; Add roles to an identity related to the account. Adding a role that is already
+; set to an identity does nothing. The only role that must be defined all the
+; time is the "owner". Other roles can be added by features and can be listed by
+; calling the `account.listRoles` endpoint.
+account.addRoles@param = {
+    ; The account to add the role to.
+    0 => account-identity,
+
+    ; A map of identity with roles to add. These roles will be added to the
+    ; identity, not replacing the ones it currently have.
+    ? 1 => {
+        * non-anonymous-identity => [ * account-role ],
+    }
+}
+account.addRoles@return = ()
+; end::addRoles[]
+
+; tag::removeRoles[]
+; Remove roles from an identity. If the role isn't set, this will do nothing.
+account.removeRoles@param = {
+    ; The account to add the role to.
+    0 => account-identity,
+
+    ; A map of identity with roles to remove from them.
+    ? 1 => {
+        * non-anonymous-identity => [ * account-role ],
+    }
+}
+account.removeRoles@return = ()
+; end::removeRoles[]
+
+; tag::info[]
+; Get the information of an account. Some information might be missing if the
+; server desires it. For example, the server might restrict which identities
+; can have the list of owners.
+account.info@param = {
+    ; The identity of the account to request the information of.
+    0 => account-identity,
+}
+account.info@return = {
+    ; The name or description of the wallet.
+    ? 0 => tstr,
+
+    ; The list of roles for the account.
+    ? 1 => {
+        * non-anonymous-identity => [ * account-role ],
+    },
+
+    ; The list of features and their argument.
+    ? 2 => [ 1* account-feature ],
+}
+; end::info[]
+
+; tag::delete[]
+; Delete an account. Its identity cannot be reused.
+; The server MUST return an error if the sender does not have the `"owner"` role
+; for this account.
+account.delete@param = {
+    0 => account-identity,
+}
+account.delete@return = ()
+; end::delete[]
+
+; tag::addFeatures[]
+; Add features and roles to an account, in one atomic call.
+; The roles added by this call can only come from the new features
+; added in this call.
+; The server MUST return an error if roles added by this call aren't available
+; to the features listed.
+; The server MAY return an error if adding features to an account is not
+; supported.
+account.addFeatures@param = {
+    ; The account to add features to.
+    0 => account-identity,
+
+    ; List of additional roles to add.
+    ? 1 => {
```

Comment:
I guess it makes sense to have it optional here, since you could add features devoid of role.